main/src/library/Fixpoint/Interpreter.flix

```diff
@@ -23,13 +23,15 @@ namespace Fixpoint {
     type alias SearchEnv[v] = (Array[Tuple[v]], Array[v])
 
     def interpret(stmt: RamStmt[v]): Database[v] & Impure with Order[v], Eq[v], ToString[v] =
+        interpretWithDatabase(MutMap.empty(), stmt)
+
+    def interpretWithDatabase(db: Database[v], stmt: RamStmt[v]): Database[v] & Impure with Order[v], Eq[v], ToString[v] =
         let _ = notifyPreInterpret(stmt);
-        let db = MutMap.empty();
         evalStmt(db, stmt);
         db
 
     def evalStmt(db: Database[v], stmt: RamStmt[v]): Unit & Impure with Order[v], Eq[v], ToString[v] = match stmt {
-        case RamStmt.Facts(fs) =>
+        case RamStmt.Facts(fs) => // TODO: consider removing this.
```

Comment:
With this change, Datalog programs should no longer contain many facts. A program with many facts is represented as `Join(Model, Datalog)` where `Model` contains the facts and `Datalog` contains the rules. So it should no longer be a performance problem to handle facts by compiling them to `project (...) into A` statements.